### ISSUE TITLE
Kotlin Code generator generates invalid code for embeddables in POJOs when immutable interfaces is activated

### ISSUE THREAD

**Issue by F43nd1r** [jOOQ/jOOQ#14853]
### Expected behavior

Generated kotlin POJOs compile when using embeddables, e.g.
```kotlin
latestVersionKey: VersionKey
    get(): VersionKey = VersionKey(
        latestVersionCode,
        latestVersionFlavor
    )
```

### Actual behavior

Generated kotlin POJOs contain invalid code, e.g.
```kotlin
 // property name missing here
    get(): VersionKey = VersionKey(
        latestVersionCode,
        latestVersionFlavor
    )
```

### Steps to reproduce the problem

Used configuration:
```kotlin
isImmutableInterfaces = true
isKotlinNotNullInterfaceAttributes = true

isKotlinNotNullRecordAttributes = true

isPojos = true
isPojosAsKotlinDataClasses = true
isKotlinNotNullPojoAttributes = true
isImmutablePojos = true
```
```kotlin
EmbeddableDefinitionType().apply {
    name = "VERSION_KEY"
    referencingName = "LATEST_VERSION_KEY"
    tables = "BUG"
    fields = listOf(
        EmbeddableField().withName("CODE").withExpression("LATEST_VERSION_CODE"),
        EmbeddableField().withName("FLAVOR").withExpression("LATEST_VERSION_FLAVOR"),
    )
}
```

### jOOQ Version

3.18.1

### Database product and version

mysql 8.0.31

### Java Version

openjdk version "17.0.6" 2023-01-17

### OS Version

Arch Linux

### JDBC driver name and version (include name if unofficial driver)

com.mysql:mysql-connector-j:8.0.32

**Comment by lukaseder** [comment]
I can reproduce it. Seems to be related to specifying immutable objects

**Comment by lukaseder** [comment]
I've fixed this for jOOQ 3.19.0 for now. Will backport soon.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.19.0, 3.18.2 (https://github.com/jOOQ/jOOQ/issues/14862), 3.17.11 (https://github.com/jOOQ/jOOQ/issues/14863), and 3.16.17 (https://github.com/jOOQ/jOOQ/issues/14864).

Thanks again for your report!
